articles/client-platforms/react/08-calling-apis.md

```diff
@@ -0,0 +1,261 @@
+      return response
+    } else {
+      var error = new Error(response.statusText)
+      error.response = response
+      throw error
+    }
+  }
+
+  fetch(url, options){
+    // performs api calls sending the required authentication headers
+    const headers = {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    }
+    // if logged in, includes the authorization header
+    if (this.loggedIn()){
+      headers['Authorization'] = 'Bearer ' + this.getToken()
+    }
+
+    return fetch(url, {
+      headers,
+      ...options
+    })
+    .then(this._checkStatus) // to raise errors for wrong status
+    .then(response => response.json()) // to parse the response as json
+  }
+}
+```
+
+As you see, there is a new `fetch` method created to send requests to private endpoints. As the native `fetch` always resolves the returned promise even for response status 401 (unauthorized), we're also adding a method to the promise stack that throws an `Error` for that case: `_checkStatus`.
+
+## 2. Create a Simple Server
+
+To exemplify how server would handle public and private endpoints, we're introducing a simple `node.js` server based on [`express`](https://expressjs.com/) and [`express-jwt`](https://github.com/auth0/express-jwt). It's very basic server with only two endpoints: `/api/public` and `/api/private`:
+
+
+```javascript
+/* ===== ./server.js ===== */
+var express = require('express');
+var app = express();
+var jwt = require('express-jwt');
+require('dotenv').config();
+
+var authenticate = jwt({
+  secret: new Buffer(process.env.AUTH0_SECRET, 'base64'),
+  audience: process.env.AUTH0_CLIENT_ID
+});
+
+app.get('/api/public', function(req, res) {
+  res.json({ message: "Hello from a public endpoint! You don't need to be authenticated to see this." });
+});
+
+app.get('/api/private', authenticate, function(req, res) {
+  res.json({ message: "Hello from a private endpoint! You DO need to be authenticated to see this." });
+});
+
+app.listen(3001);
+console.log('Listening on http://localhost:3001');
+```
+
+Both endpoints send a json response with a message attribute, but `/api/private` is using the __authenticate__ callback to validate the token received in `Authentication` header. `express-jwt` is responsible to parse and validate the token (please, check [`express-jwt` documentation for more details](https://github.com/auth0/express-jwt)). Notice we're using `dotenv` package to load `process.env.AUTH0_SECRET` and `process.env.AUTH0_CLIENT_ID` from `.env` file.
+
+<% if (account.userName) { %>
+If you don't have a `.env` file in your project you can create your own using the following command:
+
+```bash
+$ echo "AUTH0_CLIENT_ID='${account.clientId}\nAUTH0_DOMAIN='${account.namespace}'\nAUTH0_SECRET='account.secret'" > .env
+```
+<% } else { %>
+If you don't have a `.env` file in you project, please copy `.env.example` and add your own credentials from [dashboard](https://manage.auth0.com).
```

Comment:
We do not hard-code values like the dashboard URL. Instead we use variables where applicable, in this case you have to use the variable `uiURL` that holds the value `https://manage.auth0.com`.

So instead of `[dashboard](https://manage.auth0.com)`
you should use `[dashboard](${uiURL})`

Please make the update.

For more information: https://github.com/auth0/docs/blob/master/CONTRIBUTING.md#document-variables